src/NServiceBus.Core/MessageInterfaces/MessageMapper/Reflection/MessageMapper.cs

```diff
@@ -150,6 +150,14 @@ void InitType(Type t)
                 return;
             }
 
+            var typeName = GetTypeName(t);
+
+            if (!t.IsInterface)
+            {
+                nameToType[typeName] = t.TypeHandle;
```

Comment:
https://github.com/Particular/NServiceBus/pull/4133/commits/8bfd02e07d327ce1101e2fa260b258bf4fd0f919 should address this, please take a look